### PR TITLE
BUG: Timedelta(td64_out_of_bounds) silently overflowing

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -169,6 +172,69 @@ cpdef int64_t delta_to_nanoseconds(delta) except? -1:
...
+        return "W"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_M:
+        return "M"
+    elif unit == NPY_DATETIMEUNIT.NPY_FR_Y:
+        return "Y"
+    else:
+        raise NotImplementedError(unit)
+
+
+cdef object ensure_td64ns(object ts):
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx:201]: isn't this duplicating: https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/tslibs/conversion.pyx#L253

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx:201]: that is for ndarray, not td64 scalar.  i havent found a good way to share that

**Comment by jbrockmendel** [comment]
Updated to avoid using cast_from_unit (which goes through python-space)

working on a separate branch to implement tslibs.units that will, among other things, try to share the ndarray/scalar ensure-nanos functions

File: pandas/tests/io/test_html.py [hunk 2]
```diff
@@ -130,6 +130,7 @@ def test_to_html_compat(self):
         tm.assert_frame_equal(res, df)
 
     @tm.network
+    @pytest.mark.xfail(reason="2021-01-05 HTML layout changed?")
```
Comment (jreback) [inline pandas/tests/io/test_html.py:133]: these can be removed

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm pls merge master (and remove html xfails), they are on master now

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
updated + green

**Pull request merged**